tmail-backend/webadmin/webadmin-team-mailboxes/src/main/java/com/linagora/tmail/webadmin/TeamMailboxManagementRoutes.java

```diff
@@ -142,7 +151,10 @@ public Route addTeamMailbox() {
                     .haltError();
             }
 
-            return Mono.from(teamMailboxRepository.createTeamMailbox(new TeamMailbox(domain, teamMailboxName)))
+            TeamMailbox teamMailbox = new TeamMailbox(domain, teamMailboxName);
+
+            return Mono.from(teamMailboxRepository.createTeamMailbox(teamMailbox))
+                .then(Mono.from(contactSearchEngine.index(domain, new ContactFields(teamMailbox.asMailAddress(), teamMailboxName.asString(), ""))))
```

Comment:
Please do not leak logic in the presentation layer, this is a clearly wrong design pattern.

If later we add another way to add team mailboxes then the code will be duplicated at best. At worst we will have a bug.

We need IMO a generic way to extend the behavior of `teamMailboxRepository`, for instance:

```
interface TeamMailboxRepository {
    Mono<Void> teamMailboxesAdded(TeamMailbox teamMailbox);

    Mono<Void> teamMailboxesRemoved(TeamMailbox teamMailbox);
}
```

And add an implementation for the auto-complete....